src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -18,8 +19,8 @@ public class DependencyGraphSpec
     {
         private const string DGSpecFileNameExtension = "{0}.nuget.dgspec.json";
 
-        private readonly SortedSet<string> _restore = new SortedSet<string>(PathUtility.GetStringComparerBasedOnOS());
-        private readonly SortedDictionary<string, PackageSpec> _projects = new SortedDictionary<string, PackageSpec>(PathUtility.GetStringComparerBasedOnOS());
+        private readonly ConcurrentDictionary<string, bool> _restore = new ConcurrentDictionary<string, bool>();
+        private readonly ConcurrentDictionary<string, PackageSpec> _projects = new ConcurrentDictionary<string, PackageSpec>(PathUtility.GetStringComparerBasedOnOS());
```

Comment:
Why does _restore not use a comparer, while _projects does?

Reply:
I think both should use the comparer as the previous implementation did.

Reply:
Good catch. Fixed.